src/System/BpmControl.cpp

```diff
@@ -0,0 +1,37 @@
+#include "HardwareInput.h"
+#include "BpmControl.h"
+#include "ErrorLogger.h"
+
+#define ANALOG_MAX 1023
+#define ANALOG_FREQ 20000
+
+BpmControl::BpmControl(pin_size_t pinNum){
+    _pinNum = pinNum;
+}
+
+void BpmControl::init(){
+    pinMode(_pinNum, OUTPUT);
+    analogWriteRange(ANALOG_MAX);
+    analogWriteFreq(ANALOG_FREQ);
+};
+
+void BpmControl::setControlSignal(int32_t controlSignal){ 
+    if (_active == true){
+        if (controlSignal >= 0 && controlSignal <= 1023){
+            analogWrite(_pinNum, controlSignal);
+        }
+        else {
+            dyno_assert(controlSignal >= 0 && controlSignal <= 1023);
```

Comment:
asserts shouldn't be placed in an if/else block. the whole idea is that you can compile them out if you want to (when building for production, for example). if you do that in this case then you get an empty else condition.

I would just place dyno_assert before the call to analogWrite

Reply:
I recommend:

```c++
// ...
    if (_active = true){
        dyno_assert(controlSignal >= 0 && controlSignal <= 1023);
        analogWrite(_pinNum, controlSignal);
    } else {
    // ...
```